### PR TITLE
PR for Statement of Accounts

### PR DESCRIPTION
**Pull request by jiloysss**
Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

SAMPLE GENERATED REPORT
image

SAMPLE PRINT FORMAT FOR THE GENERATED REPORT
image

### CODE REVIEW

File: erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+            for i in range(0,len(data[0])):
+                if isinstance(i, int):
+                    final_data[0].append(0)
+                elif isinstance(i, float):
+                    final_data[0].append(0.0)
+                else:
+                    final_data[0].append(None)
+
+            for x in data:
+                print("XXXXXXXXXXXXXXXXXXXXXXXXXXx")
```
Comment (deepeshgarg007) [inline erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py]: Please remove print statements

File: erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+                    # invoiced and paid amounts
+                    invoiced_amount = gle.get(dr_or_cr) if (gle.get(dr_or_cr) > 0) else 0
+                    paid_amt = invoiced_amount - outstanding_amount
+                    row += [invoiced_amount, paid_amt, outstanding_amount]
+
+                    # ageing data
+                    entry_date = due_date if self.filters.ageing_based_on == "Due Date" else gle.posting_date
+                    row += get_ageing_data(cint(self.filters.range1), cint(self.filters.range2),
+                                           cint(self.filters.range3), self.age_as_on, entry_date, outstanding_amount)
+                    print("ROOOOOOOOOOOW")
```
Comment (deepeshgarg007) [inline erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py]: Remove print statements

File: erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+                    print(row)
+                    # issue 6371-Ageing buckets should not have amounts if due date is not reached
+                    if self.filters.ageing_based_on == "Due Date" and getdate(due_date) > getdate(
+                            self.filters.report_date):
+                        row[-1] = row[-2] = row[-3] = row[-4] = 0
+
+                    #if self.filters.get(scrub(args.get("party_type"))):
```
Comment (deepeshgarg007) [inline erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py]: Remove commented code

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

**Comment by jaywks** [comment]
What is needed to merge this?

File: erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+				from `tabGL Entry`
+				where docstatus < 2 and party_type=%s and (party is not null and party != '')
+				and (party='{1}' or against='{1}') {2}
+				group by voucher_type, voucher_no, against_voucher_type, against_voucher, party
+				order by posting_date, party"""
+                                            .format(select_fields, self.filters.get("party"),conditions), values, as_dict=True)
+
+        return self.gl_entries
+
+    def prepare_conditions(self, party_type):
```
Comment (rmehta) [inline erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py:261]: Indentation using tabs

https://github.com/frappe/erpnext/wiki/Coding-Standards#use-simple-indenting

File: erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+        """final_data = []
+        size_data = 0
+        if data:
+            size_data = len(data)
+            final_data.append(data[size_data-1])"""
+
+        return columns, final_data, None, chart
+
+    def get_columns(self, party_naming_by, args):
+        columns = [_("Posting Date**") + ":Date:80",
```
Comment (rmehta) [inline erpnext/accounts/report/statement_of_accounts_receivable/statement_of_accounts_receivable.py:50]: please use new JSON format for column definitions

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by jaywks** [comment]
@jiloysss please address the changes requested thank you very much for your effort.

**Pull request closed without merging**